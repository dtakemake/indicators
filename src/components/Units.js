import React from 'react'

import {
  ButtonGroup,
  Button
} from 'react-bootstrap'

const Units = ({ 
  
  /* 
  * @param { key : value } доступные единицы размера
  */
  units,

  /*
  * @param string { key } текущая единица размера
  */
  unit, 

  /*
  * @param function изменить единицу измерения
  */
  setUnit, 

  /* 
  * @param string - класс кнопок
  */
  variant = "secondary"

}) => {
  // выберем все ключи измерений
  const keys = Object.keys(units)

  return keys ? 
    <ButtonGroup size="sm">
      {keys.map(key => key !== unit ?
        <Button variant={variant} key={key} onClick={() => setUnit(key)}>{units[key]}</Button>
        :
        <Button variant={variant} key={key} disabled="disabled">{units[key]}</Button>)}
    </ButtonGroup>
    :
    null
  }

export default React.memo(Units)