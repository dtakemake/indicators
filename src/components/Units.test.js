import React from 'react'
import ReactDOM from 'react-dom'
import TestRenderer from 'react-test-renderer'
import Units from './Units'

describe('Units', () => {
 
  let props = {
    units: {
      "kg" : "кг", 
      "lb" : "фунты"
    }, 
    unit: "kg",
    setUnit() {}, 
    variant: "warning"
  }
  
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Units { ...props } />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  test('есть корректный снимок', () => {
    const component = TestRenderer.create(<Units { ...props } />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})