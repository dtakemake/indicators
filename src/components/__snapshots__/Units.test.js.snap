// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Units есть корректный снимок 1`] = `
<div
  className="btn-group btn-group-sm"
  role="group"
>
  <button
    className="btn btn-warning"
    disabled="disabled"
    type="button"
  >
    кг
  </button>
  <button
    className="btn btn-warning"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    фунты
  </button>
</div>
`;
